### PR TITLE
ovnkube.sh: optionally run as non-root user

### PR DESCRIPTION
**Pull request by orgcandman**
When the shell script sees the OVS_USER_ID parameter, it will run the
various daemons as that user.  The user-info is specified in the standard
user:group notation.

As an implementation detail, using libcap-ng to do the domain transition for
the ovn containers seemed to cause issues, possibly with the capability
bits being retained.  For now, use setpriv as a workaround, but it could be
addressed in the future without impacting users.

Signed-off-by: Aaron Conole <aconole@redhat.com>

### CODE REVIEW

**Comment by orgcandman** [comment]
@danwinship  and @dcbw - please check for coding style, project fitness, etc.

**Comment by dcbw** [comment]
@girishmg @shettyg what do you think?

File: dist/images/ovnkube.sh [hunk 1]
```diff
@@ -649,6 +685,7 @@ ovn-northd () {
 # make sure the specified dport is open
 iptables-rules () {
   dport=$1
+
```
Comment (dcbw) [inline dist/images/ovnkube.sh]: @orgcandman seems unrelated?

Comment (orgcandman) [reply dist/images/ovnkube.sh]: Sure - want a push with it removed?

**Comment by girishmg** [comment]
@orgcandman Have you tried this using Ubuntu container image?

**Comment by orgcandman** [comment]
I haven't.  Do you suspect a problem?

**Comment by girishmg** [comment]
OVS/OVN packages in ubuntu are not build with libcap-ng. It wasn't clear from your PR description  whether you were dependent on it or not. If you are dependent on it, then things may not work on ubuntu.

**Comment by orgcandman** [comment]
It shouldn't matter which mechanism is used.  The ability to set uid / gid was present before for the OVS daemons.   This commit doesn't change that.  It only extends that capability to the ovn daemons (using setpriv rather than --ovn-user).

File: dist/images/ovnkube.sh (hunk 1 above)
Comment (dcbw) [reply dist/images/ovnkube.sh]: yes if you respin, otherwise no

**Comment by dcbw** [comment]
@girishmg tested in vagrant with an Ubuntu image:

```
ovn-kubernetes   ovnkube-db-68987d44f4-bgrq2         2/2     Running            0          17m
ovn-kubernetes   ovnkube-master-68df776b69-bs7jk     2/2     Running            0          17m
ovn-kubernetes   ovnkube-node-8x2ct                  3/3     Running            0          13m
ovn-kubernetes   ovnkube-node-j7p7w                  3/3     Running            0          17m
ovn-kubernetes   ovnkube-node-m74vb                  3/3     Running            0          9m2s
```
